File: src/csharp/core/src/Style/ComponentTranslate.cs
using System.Text.Json.Nodes;

namespace DiceBear.Internal
{
    /// <summary>
    /// Read-only view over a component's <c>translate</c> block, providing the
    /// X and Y offset ranges.
    /// </summary>
    internal sealed class ComponentTranslate
    {
        private readonly JsonObject _data;

        internal ComponentTranslate(JsonObject data) => _data = data;

        internal NumberRange? X() => JsonRead.Range(_data, "x");

        internal NumberRange? Y() => JsonRead.Range(_data, "y");
    }
}

File: src/go/core/internal/style/component_translate.go
package style

import "github.com/dicebear/dicebear-go/v11/internal/prng"

// componentTranslate is a read-only view over a component's translate block,
// providing the X and Y offset ranges.
type componentTranslate struct {
	X *prng.Range `json:"x"`
	Y *prng.Range `json:"y"`
}
